contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1926,11 +1927,22 @@ pltsql_store_view_definition(const char *queryString, ObjectAddress address)
 	flag_validity |= BBF_VIEW_DEF_FLAG_USES_QUOTED_IDENTIFIER;
 	if (pltsql_quoted_identifier)
 		flag_values |= BBF_VIEW_DEF_FLAG_USES_QUOTED_IDENTIFIER;
+	/*
+	 * Setting this flag bit to 0 to distinguish between the objects
+	 * created in 2.x or 3.x for future references. Let's not use
+	 * this bit in 3.x, as we are setting this to 1 in 2.x and will
+	 * be reserved for MVU.
+	 */
+	flag_validity |= BBF_VIEW_DEF_FLAG_CREATED_IN_OR_AFTER_2_4;
```

Comment:
why setting it to 0? why not set it to 1

Reply:
This is to distinguish between 2.4 and 3.1 versions in 3.2, if we want to introduce a new bit in future.
We are keeping 3rd bit from the right to track the change from 2.x, explicitly keeping it 0 in 3.x.
When we upgrade from 2.4 -> 3.1 -> 3.2, the 3rd bit will be 1 which means the object was created in 2.x.

Reply:
1 does not mean it is from 2.x, 1 means it is from 2.4 while 0 means it is from 2.3. Now with this change, 1 means it is from 2.4 and 0 means it is from 2.3/3.x which is confusing. Otherwise this could be used by customer to find all the views which needs recreation to get correct definition in the catalog (views with bit set to 0)

Reply:
I agree. But this bit is not added for the customer to find out the views which needs to be recreated, it is solely for the MVU purpose. This is an internal table.

We can convey to customer "If the view definition comes out be NULL, then it needs to be recreated".

NOTE: If the customer is on 3.x,  the view definition would not be NULL even though the 3rd bit will be 0 except 3.0.
1. 2.3 -> 3.0 (create V1) -> 3.1 ( the view definition V1 would be NULL) which means that the view V1 needs to be recreated.